### ISSUE TITLE
Phase 5 finding: top-level dxs_dE / dxs_dmu / ddxs internal integration paths not fully tracer-safe

### ISSUE THREAD

**Issue by gschnabel** [IAEA-NDS/endf-userpy#220]
Discovered by Phase 5 cross-MF end-to-end grad pins (roadmap #198).

## Summary

Four of the five top-level user-facing entry points in \`endf_userpy.quantities\` currently support \`jax.grad\` end-to-end:

| entry point                            | grad wrt E | grad wrt E' | grad wrt mu |
|:---------------------------------------|:----------:|:-----------:|:-----------:|
| \`get_reaction_xs\`                    | working    | -           | -           |
| \`get_particle_production_xs\`         | working    | -           | -           |
| \`get_particle_production_dxs_dE\`     | **broken** | **broken**  | -           |
| \`get_particle_production_dxs_dmu\`    | **broken** | -           | **broken**  |
| \`get_particle_production_ddxs\`       | working    | **broken**  | **broken**  |

## Root cause

The four broken paths share a common structure: internal mu-quadrature (\`integrate_mf6_dist2d_over_mu\`) or Ep-quadrature (\`integrate_mf6_dist2d_over_eout\`) built with \`xp.linspace\` inside the traced computation. Under \`jax.grad\`, that quadrature is a jax tracer even when the physical mu / Ep values are constant. When passed down into LAW=1 / LAW=7 kernels that internally do \`np.asarray(mu_out)\` / \`np.asarray(ep_out)\`, the tracer materialises and crashes.

Concrete failure trace (Be-9 MT=16, LAW=7, \`get_particle_production_ddxs\` grad wrt mu):

\`\`\`
mf6_interpretation_subsecs.py:636, in _get_dist2d_from_subsec_law7_traced_x:
    f1 = _law7_eval_mu_panel(subsec, p + 1,
                              np.asarray(mu_out, dtype=float),  # <-- tracer
                              np.asarray(ep_out, dtype=float),
                              xp)
TracerArrayConversionError
\`\`\`

## What works

- Standalone LAW=7 traced-x kernel with concrete numpy mu / eout: works (\`tests/test_mf6_law7_grad_wrt_E.py\`).
- Standalone LAW=1 multipanel-traced kernel: works (\`tests/test_mf6_law1_multipanel_traced.py\`).
- End-to-end \`get_reaction_xs\` grad wrt E (no distribution routing): works.
- End-to-end \`get_particle_production_xs\` and \`get_particle_production_ddxs\` grad wrt E: works.

The gap is specifically in the mu / Ep integration layer between the top-level API and the leaf kernels.

## Fix directions

**Approach A (surgical)**. Make the mu / Ep integration primitives xp-agnostic on the quadrature-node axis, so they pass concrete numpy nodes to the leaf kernels regardless of the outer \`xp\`. Requires threading \`xp\` through \`integrate_mf6_dist2d_over_mu\` / \`_over_eout\` and making the leaf kernel accept an explicit "use numpy for the mu/Ep axis" hint. The trickier part is preserving grad wrt the query axis; the quadrature nodes themselves are non-differentiable file-side data, so this is doable.

**Approach B (pcb wrap)**. Wrap the whole mu / Ep integration in \`jax.pure_callback\`. Preserves finite gradient rules manually. More invasive but keeps end-to-end xp=jax semantics.

**Approach C (structural)**. Add unit-base support (INT=21..25) to \`_interp_tab2_traced_x\` so LAW=7 \`_law7_eval_mu_panel\` can run tracer-natively. Also add multi-panel support with kink-aware kinematics for LAW=1 mu-quadrature. Largest scope.

Approach A is probably the smallest deliverable per the "primitive first" pattern established by #201.

## Test pins landing with this issue

\`tests/test_phase5_end_to_end_grad.py\` includes six \`pytest.skip\` markers on the failing configurations that document the current state so the fix flips them to positive assertions later.
